charts/px-backup/templates/pxcentral-pre-install-hook.yaml

```diff
@@ -0,0 +1,77 @@
+{{/* Setting defaults if they are omitted. */}}
+{{- $customRegistryEnabled := .Values.customeRegistryEnabled | default false }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pxcentral-onprem-job
+  namespace: {{ .Values.namespace }}
+  annotations:
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+  labels:
+{{- include "px-backup.labels" . | nindent 4 }}
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pxcentral-onprem-job
+  namespace: {{ .Values.namespace }}
+  annotations:
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-weight": "-5"
+  labels:
+{{- include "px-backup.labels" . | nindent 4 }}
+rules:
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs: ["*"]
```

Comment:
does this job required all these permissions?

Typically enterprise customers aren't font of such clusterroles that take all privileges.

Reply:
Changed.